stepsfunction/handler.py

```diff
@@ -163,19 +178,186 @@ def ocr_page(event, context):
     See Callback Pattern at https://console.aws.amazon.com/states/home?region=us-east-1#/sampleProjects
     """
     LOG.info(f'event: {dumps(event)}')
-    activity_arn = os.environ.get('ACTIVITY_ARN', "")
+    LOG.info(f'boto3 version: {boto3.__version__} botocore version: {botocore.__version__}')
+
     s3rec = event['Records'][0]['s3']  # only the first, but there should only be one for S3
     bucket = s3rec['bucket']['name']
     key = s3rec['object']['key']
     _doc_pdf, jid, name_pdf = key.split('/')
     LOG.info(f'bucket={bucket} key={key} jid={jid} name_pdf={name_pdf}')
 
-    worker_name = "notify_upload_worker"
-    sf = boto3.client('stepfunctions')
+    detected_text = get_textract_data(bucket, key)
+    write_textract_to_s3(detected_text, bucket, key)
+    pdf_upload = PDFUpload.get(hash_key=jid)
+    update_db_after_ocr(pdf_upload, detected_text, name_pdf, jid)
+
     # Pretend we've discovered we've finished all the pages
-    if name_pdf == '0000.pdf':
-        res = sf.get_activity_task(activityArn=activity_arn, workerName=worker_name)
-        LOG.info(f'sf_activity={res}')
-        sf_input = dumps(res['input'])
-        res = sf.send_task_success(taskToken=res['taskToken'], output=sf_input)
+    if check_orc_done(pdf_upload, jid):
+        # update db
+        pdf_upload.update_with_log(
+            actions=[
+                PDFUpload.status.set("ocred"),
+                PDFUpload.updatedAt.set(datetime.now())
+            ])
+        send_task_ocr_activity(jid)
+
+
+def get_textract_data(bucket, key):
+    """Using AWS textract."""
+    LOG.info(f'Loading get_textract_data bucket:{bucket}, key:{key}')
+    # convert to jpg
+    jpg_file = extract_jpg_from_pdf(bucket, key)
+    # Call Amazon Textract
+    tx = boto3.client(
```

Comment:
Probably best to do this at the top of the file, so the Client is created the first time the Lambda is invoked, in the "cold start". That way it doesn't have to do it again when the same lambda engine is called alter on a "warm start". Anything that's time or CPU expensive should be before the handler is invoked.